Source/JavaScriptCore/runtime/JSObjectInlines.h

```diff
@@ -206,6 +206,32 @@ inline JSValue JSObject::getIfPropertyExists(JSGlobalObject* globalObject, Prope
     return slot.getValue(globalObject, propertyName);
 }
 
+// FIXME: Given the single special purpose this is used for, it's unclear if this needs to be a JSObject member function.
+inline bool JSObject::noSideEffectMayHaveNonIndexProperty(VM& vm, PropertyName propertyName)
+{
+    // This function only supports non-index PropertyNames.
+    ASSERT(!parseIndex(propertyName));
+    ASSERT(propertyName != vm.propertyNames->length);
+    for (auto* object = this; object; object = object->getPrototypeDirect().getObject()) {
+        auto inlineTypeFlags = object->inlineTypeFlags();
+        if (TypeInfo::overridesGetOwnPropertySlot(inlineTypeFlags) && object->classInfo() != ArrayPrototype::info())
```

Comment:
Would you like to exclude JSArray::info() as well?

Reply:
I am willing to add that in the future if it’s helpful. For the moment, there is no need for that additional check.

This is part of the tension for this function; is it general purpose or just for one particular use in `fastStringify`? At the moment there is no real benefit to adding a check for `JSArray` given how this is used.